src/ripple/app/tx/impl/Escrow.cpp

```diff
@@ -105,8 +107,18 @@ EscrowCreate::preflight(PreflightContext const& ctx)
     if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
         return ret;
 
-    if (!isXRP(ctx.tx[sfAmount]))
-        return temBAD_AMOUNT;
+    STAmount const amount{ctx.tx[sfAmount]};
+    if (!isXRP(amount))
+    {
+        if (!ctx.rules.enabled(featurePaychanAndEscrowForTokens))
+            return temBAD_AMOUNT;
+
+        if (!isLegalNet(amount))
+            return temBAD_AMOUNT;
+
+        if (isFakeXRP(amount))
+            return temBAD_CURRENCY;
```

Comment:
The method was created because the same comparison was done in 3 locations. 3 is usually the number of times I repeat something before I make a new method/function. We can just do a comparison if that is recomended.

Reply:
Yeah I think having a "cleanup" pr after is a good approach and I don't mind making that part of our process. I've seen a few instances now where it feels right to update the rest of the code to match your awesome new function but adding those files to the commit history feels wrong because its not "required" for this amendment to operate.